packages/synapse-react-client/src/components/FullTextSearch/FullTextSearchUtils.ts

```diff
@@ -40,7 +40,11 @@ export function getTextMatchesQueryFilter(
   const { textMatchesMode, distance = 0 } = ftsConfig
   let searchExpression = searchText
   if (textMatchesMode == 'BOOLEAN') {
-    const searchTextWordLength = searchText.trim().split(/\s+/).length
+    //split by non-word character and ignore empty strings
+    const searchTextWordLength = searchText
+      .trim()
+      .split(/\W+/)
+      .filter(word => word.length > 0).length
```

Comment:
Probably reuse the `getWordCount` in `TextFieldWithWordLimit.tsx` (maybe move that to a generic utils module)